conjure-java-core/src/test/java/com/palantir/conjure/java/UndertowServiceEteTest.java

```diff
@@ -389,75 +374,57 @@ public void testOptionsOnGetIncludesHead() throws IOException {
     }
 
     @Test
     public void testPlusInPathParam() {
         String expected = "foo+bar";
         assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
-        assertThat(Futures.getUnchecked(retrofitClient.path(AuthHeader.valueOf("bearer"), expected)))
-                .isEqualTo(expected);
+        assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
     }
 
     @Test
     public void testSpaceInPathParam() {
         String expected = "foo bar";
         assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
-        assertThat(Futures.getUnchecked(retrofitClient.path(AuthHeader.valueOf("bearer"), expected)))
-                .isEqualTo(expected);
+        assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
     }
 
     @Test
     public void testBinaryOptionalEmptyResponse() {
-        Optional<ResponseBody> response =
-                Futures.getUnchecked(binaryClient.getOptionalBinaryEmpty(AuthHeader.valueOf("authHeader")));
+        Optional<InputStream> response = binaryClient.getOptionalBinaryEmpty(AuthHeader.valueOf("authHeader"));
         assertThat(response).isNotPresent();
     }
 
     @Test
     public void testBinaryOptionalPresentResponse() throws Exception {
-        Optional<ResponseBody> response =
-                Futures.getUnchecked(binaryClient.getOptionalBinaryPresent(AuthHeader.valueOf("authHeader")));
+        Optional<InputStream> response = binaryClient.getOptionalBinaryPresent(AuthHeader.valueOf("authHeader"));
         assertThat(response).isPresent();
-        assertThat(response.get().string()).isEqualTo("Hello World!");
+        assertThat(new String(response.get().readAllBytes(), StandardCharsets.UTF_8))
+                .isEqualTo("Hello World!");
     }
 
     @Test
+    @SuppressWarnings("MustBeClosedChecker")
     public void testBinaryServerSideFailureAfterManyBytesSent() {
-        ResponseBody response = Futures.getUnchecked(binaryClient.getBinaryFailure(
+        InputStream response = binaryClient.getBinaryFailure(
                 AuthHeader.valueOf("authHeader"),
                 // Write more bytes than one buffer
-                20000));
-        assertThatThrownBy(() -> response.byteStream().transferTo(ByteStreams.nullOutputStream()))
-                .isInstanceOf(IOException.class);
+                20000);
+        assertThatThrownBy(response::readAllBytes).isInstanceOf(IOException.class);
     }
 
     @Test
+    @SuppressWarnings("MustBeClosedChecker")
     public void testBinaryServerSideFailureAfterFewBytesSent() {
         assertThatThrownBy(() -> {
                     try {
-                        Futures.getUnchecked(binaryClient.getBinaryFailure(AuthHeader.valueOf("authHeader"), 1));
+                        binaryClient.getBinaryFailure(AuthHeader.valueOf("authHeader"), 1);
```

Comment:
with a small modification we can avoid suppressions
```suggestion
                        binaryClient.getBinaryFailure(AuthHeader.valueOf("authHeader"), 1).close();
```